devices/YarpRobotLoggerDevice/tests/YarpRobotLoggerDeviceTest.cpp

```diff
@@ -0,0 +1,233 @@
+/**
+ * @copyright 2024 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#include <cstdlib>
+#include <chrono>
+#include <filesystem>
+#include <random>
+#include <thread>
+
+// Catch2
+#include <catch2/catch_test_macros.hpp>
+
+// YARP
+#include <yarp/sig/Vector.h>
+#include <yarp/eigen/Eigen.h>
+#include <yarp/os/Bottle.h>
+#include <yarp/os/BufferedPort.h>
+#include <yarp/os/Network.h>
+#include <yarp/robotinterface/XMLReader.h>
+#include <yarp/dev/IEncoders.h>
+#include <yarp/dev/MultipleAnalogSensorsInterfaces.h>
+#include <yarp/dev/PolyDriver.h>
+
+// matioCpp
+#include <matioCpp/matioCpp.h>
+
+template <typename T, typename U>
+void checkVectorAreEqual(const T& vector1, const U& vector2, double tol = 0)
+{
+    REQUIRE(vector1.size() == vector2.size());
+
+    for (unsigned int i = 0; i < vector1.size(); i++)
+        REQUIRE(std::abs(vector1[i] - vector2[i]) <= tol);
```

Comment:
A possibility is to use eigen isApprox function

Reply:
Done in  https://github.com/ami-iit/bipedal-locomotion-framework/pull/862/commits/5c1861f96c0fa6e1ac669e1dd90dbde3f4adafd9.